### PR TITLE
add docs for testing pull request changes

### PR DESCRIPTION
**Pull request by yharish991**
# Description
This PR adds some documentation on how to test the pull request changes when contributing to build tooling. The docs in this PR are a good starting point and can be extended to add other scenarios later.

Fixes https://github.com/vmware-tanzu/build-tooling-for-integrations/issues/41

## Change Details
Check all that apply:
- [ ] New feature 
- [ ] Bug fix 
- [ ] Non-code change 
- [ ] Non-breaking change 
- [ ] Breaking change
- [ ] Requires a documentation change

## Release Note

```release-note
```

# How Has This Been Tested?
_Describe any testing done to verify changes. Provide enough detail that tests can be reproduced._

# Checklist:
- [x] My code follows the [Contribution and Style guidelines](../../CONTRIBUTING.md) of this project
- [x] My changes generate no new warnings
- [x] New and existing unit tests pass locally with my changes
- [x] I have added test details that prove my fix is effective or that my feature works

### CODE REVIEW

File: docs/test-pull-request-changes.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
+# Testing pull request changes
+
+This document provides some guidance to test pull request changes.
```
Comment (codegold79) [inline docs/test-pull-request-changes.md:3]:
```suggestion
This document provides guidance on testing pull request changes.
```

File: docs/test-pull-request-changes.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+Whenever the files in the [templates](../templates) directory are updated, changes need to be validated before creating
+the pull request. We can use one of the existing example projects in the [examples](../examples) directory to validate
+the changes.
+
+* When the Makefile is updated, make sure that the changes are validated by running the make targets in the example
+project.
+* When the Dockerfile is updated, go to the example project and run `make init` and copy the Dockerfile changes from
+templates directory to the Dockerfile in the example project and run the make target `docker-all` to validate that you
+are able to build the example successfully.
```
Comment (codegold79) [inline docs/test-pull-request-changes.md:15]: There are a lot of `and`s in this sentence. I would recommend writing a list separated by commas, with one `and` before the last action to do. Or break up the sentence.

File: docs/test-pull-request-changes.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+* Similarly, if you update the .golangci.yaml file, copy the changes to the .golangci.yaml file in the example project
+and validate the linter config changes.
```
Comment (codegold79) [inline docs/test-pull-request-changes.md:17]: How do you validate the linter config changes?

File: docs/test-pull-request-changes.md [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+## Testing to be done when updating package tooling image
```
Comment (codegold79) [inline docs/test-pull-request-changes.md:19]:
```suggestion
## Testing to be done when updating package tooling images
```

File: docs/test-pull-request-changes.md [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+
+Whenever the files in [package-tooling-image](../package-tooling-image) directory are updated, we need to validate the
+changes before creating the pull request. We can use the one of existing example projects in the [examples](../examples)
+directory to validate the changes.
```
Comment (codegold79) [inline docs/test-pull-request-changes.md:23]: I think `we` should be `you`. We (the build tooling devs) aren't doing the validation. Whoever is reading this file should be doing all this. Not sure who that is, but I'm sure they know.

File: docs/test-pull-request-changes.md [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+
+* Build the package tooling image by running the `build-package-tooling` make target in the build tooling project root
+directory and use that image in the Makefile of one of the example projects and validate that you are able to build
+and push the package and repo bundle in the example project. Easiest way to use and test the newly built package tooling
```
Comment (codegold79) [inline docs/test-pull-request-changes.md:27]: Again, this is a run-on sentence with too many `and`s.

Comment (codegold79) [inline docs/test-pull-request-changes.md:27]:
```suggestion
and push the package and repo bundle in the example project. The easiest way to use and test the newly built package tooling
```

**Review by codegold79 — APPROVED** [review]
Thanks for writing documentation and helping people understand how to use our tools.

File: docs/test-pull-request-changes.md (hunk 4 above)
Comment (yharish991) [reply docs/test-pull-request-changes.md:19]: there is only one package tooling image

File: docs/test-pull-request-changes.md (hunk 3 above)
Comment (yharish991) [reply docs/test-pull-request-changes.md]: expanded more to address the question

**Review by jmoroski — APPROVED** [review]

**Pull request merged**